WebAPI/Controllers/StudentController.cs

```diff
@@ -180,5 +179,61 @@ public async Task<IActionResult> AddNewBlock([FromBody]AddNewBlockModel newBlock
             }
             return Ok();
         }
+
+        [HttpPut("editBlock")]
+        public async Task<IActionResult> EditBlock([FromBody]AddNewBlockModel newBlockModel)
+        {
+            var student = await _studentService.FindByIdAsync(newBlockModel.User.Student.Id);
+
+            if (student == null)
+            {
+                return ErrorResponse($"Student with id: {newBlockModel.User.Student.Id} does not exist.");
+            }
+
+            if (student.Timetable == null)
+            {
+                return ErrorResponse($"Timetable for student with id: {student.Id} does not exist.");
+            }
+
+            Block block = new Block();
+            TimetableBlock timetableBlock = newBlockModel.TimetableBlock;
+            Course course = await _courseService.FindByNameAsync(timetableBlock.CourseName);
+            if (course == null)
+            {
+                course = new Course
+                {
+                    Id = Guid.NewGuid(),
+                    CourseName = timetableBlock.CourseName,
+                    CourseCode = "",
+                    Timetable = new FRITeam.Swapify.Entities.Timetable()
+                };
+
+                block.CourseId = course.Id;
+                block.Day = (Day)timetableBlock.Day;
+                block.StartHour = (byte)timetableBlock.StartBlock;
+                block.Duration = (byte)(timetableBlock.EndBlock - timetableBlock.StartBlock);
+                block.Room = timetableBlock.Room;
+                block.Teacher = timetableBlock.Teacher;
+                block.BlockType = (BlockType)timetableBlock.Type;
+
+                await _courseService.AddAsync(course);
+                course.Timetable.AllBlocks.Add(block);
+            }
+            else
+            {
+                block.CourseId = course.Id;
+                block.Day = (Day)timetableBlock.Day;
+                block.StartHour = (byte)timetableBlock.StartBlock;
+                block.Duration = (byte)(timetableBlock.EndBlock - timetableBlock.StartBlock);
+                block.Room = timetableBlock.Room;
+                block.Teacher = timetableBlock.Teacher;
+                block.BlockType = (BlockType)timetableBlock.Type;
```

Comment:
The same code 2x... What about a method with param timetableBlock to copy data to block?

Reply:
I resolved this in " Edit block on backend is done " commit at branch #79